onnxruntime/python/tools/symbolic_shape_infer.py

```diff
@@ -1600,6 +1601,50 @@ def _infer_ReduceSum(self, node):  # noqa: N802
                     )
                 )
 
+    def _infer_ReduceMean(self, node):  # noqa: N802
+        keep_dims = get_attribute(node, "keepdims", 1)
+        opset = get_opset(self.out_mp_)
+
+        if opset >= 13 and len(node.input) > 1:
+            axes = self._try_get_value(node, 1)
+        else:
+            axes = get_attribute(node, "axes")
+
+        vi = self.known_vi_[node.output[0]]
+
+        if axes is None:
+            assert keep_dims == 1, "ReduceMean Op: Cannot infer shape when axes is unknown and keepdims is not 1."
```

Comment:
opset 18 add an attribute `noop_with_empty_axes`, shall we handle it here.
https://onnx.ai/onnx/operators/onnx__ReduceMean.html#reducemean-18